### PR TITLE
Components: Integrate "next" Text component

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Description
Integrates the G2 text component.

## How has this been tested?
`Text` is only used in `edit-site` currently and I'm unsure how to test it, but I've added stories for the new component that test the adapter and it appears to work so I think this is safe 🙂 

## Types of changes
New feature (non-breaking change to existing Text component)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/components/src/__next/context/with-next.js [hunk 1]
```diff
@@ -31,5 +27,5 @@ export function withNext(
 		return contextConnect( WrappedComponent, namespace );
 	}
 
-	return forwardRef( CurrentComponent );
+	return CurrentComponent;
```
Comment (saramarcondes) [inline packages/components/src/__next/context/with-next.js:30]: We cannot automatically forward the ref of any component (for example, you cannot call forward ref on a styled component like `Text`).

**Comment by ItsJonQ** [comment]
@saramarcondes Thank you for working on this one 🙏 !!

I gave it some thought. I think we should just replace the current (web) `Text` component with the one from G2 Components, rather than using the `next` adapter.

Fortunately, the (current) `Text` is under `experiment` and it isn't used in a lot of places.

I poked around, and these seem to be the only places it's being used so far:

* packages/components/src/input-control/styles/input-control-styles.js
* packages/components/src/navigation/styles/navigation-styles.js
* packages/components/src/resizable-box/resize-tooltip/styles/resize-tooltip.styles.js
* packages/edit-site/src/components/template-details/index.js

I vote that we use it directly 👍
That'll allow us to remove a lot of code. I'm not even sure if we need to preserve the previous `variant` API.
The G2 `Text` has that built in as part of it's `size` style.

Let me know what you think!

If you decide to make the update, I'd be more than happy to help you refactor where it's used + test things out.

**Comment by saramarcondes** [comment]
@ItsJonQ Sounds good to me! Do you know how to test the `edit-site` package? I'm totally unaware of how to access that code in a running environment.

**Review by ItsJonQ — APPROVED** [review]
🚀 from me! Thank you @saramarcondes .

I paired with Sara this afternoon. In attempting to use only the new `<Text />` component, we ran into a potential issue with the `Navigation` component.

For that, we think it would be better to go with the adapter strategy for now. A good follow up would be to double check that using the new version of `Text` does not break integration with `Navigator`.

**Pull request merged**